app/controllers/posts_controller.rb

```diff
@@ -237,6 +238,29 @@ def destroy_many
     render nothing: true
   end
 
+  def merge_posts
+    params.require(:post_ids)
+
+    posts = Post.where(id: post_ids_including_replies).order(:id)
+    raise Discourse::InvalidParameters.new(:post_ids) if posts.blank?
+
+    # Make sure we can delete the posts
+    posts.each {|p| guardian.ensure_can_delete!(p) }
```

Comment:
Any part in specific? It's the same way as the other methods in the controller. Example: https://github.com/discourse/discourse/blob/master/app/controllers/posts_controller.rb#L162